crates/matrix-sdk-ui/src/room_list_service/room.rs

```diff
@@ -150,27 +150,32 @@ impl Room {
     }
 
     /// Create a new [`TimelineBuilder`] with the default configuration.
+    ///
+    /// If the room was synced before some initial events will be added to the
+    /// [`TimelineBuilder`].
     pub async fn default_room_timeline_builder(&self) -> Result<TimelineBuilder, Error> {
         // TODO we can remove this once the event cache handles his own cache.
 
-        let sliding_sync_room =
+        let sliding_sync_room = self
+            .inner
+            .sliding_sync
+            .get_room(self.inner.room.room_id())
+            .await
+            .ok_or_else(|| Error::RoomNotFound(self.inner.room.room_id().to_owned()));
+
+        if let Some(sliding_sync_room) = sliding_sync_room {
```

Comment:
Is this code correct? I'd think that `sliding_sync_room` is a `Result` here, not an `Option`…